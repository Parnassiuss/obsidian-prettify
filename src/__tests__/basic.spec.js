// @ts-nocheck
const Remarkable = require("remarkable");
const prettify = require("../index");

function parseContents(
    content,
    settings = {}
) {
    return new Remarkable.Remarkable()
        .use(prettify)
        .render(content);
}

describe("basic", () => {
    const content = require("../mockData");

    test("Basic functionality: tables", () => {
        expect(parseContents(
            `
## Tables

**A**|**B**|**C**
|---:|:---|---|
a |b |c
x |y |z
            `
        )).toMatchSnapshot();
    });


    test("Basic functionality: alternate lists", () => {
        expect(parseContents(
            `
## List alternation

* a
* b
  * c
  * d
    * e
    * f
            `
        )).toMatchSnapshot();
    });

    test("Basic functionality: enumerate lists", () => {
        expect(parseContents(
            `
## Lists numbering

1. foo
  a. aaa
  b. bbb
  c. ccc
1. bar
1. baz
            `
        )).toMatchSnapshot();
    });

    test("Basic functionality: handle links", () => {
        expect(parseContents(
            `
## Strange links

A link : [foo](bar){#baz}
`
        )).toMatchSnapshot();
    });


});
