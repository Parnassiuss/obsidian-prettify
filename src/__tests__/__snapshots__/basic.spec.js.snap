// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic Basic functionality: alternate lists 1`] = `
"## List alternation

* a
* b
  - c
  - d
    + e
    + f
"
`;

exports[`basic Basic functionality: enumerate lists 1`] = `
"## Lists numbering

1. foo
  a. aaa
  b. bbb
  c. ccc
2. bar
3. baz
"
`;

exports[`basic Basic functionality: handle links 1`] = `
"## Strange links

A link : [foo](bar#baz)
"
`;

exports[`basic Basic functionality: tables 1`] = `
"## Tables

| **A** | **B** | **C** | 
| ---: | :--- | --- |
| a | b | c |
| x | y | z |
"
`;
